### DISCUSSION TITLE
Import assignment cannot be used when targeting ECMAScript modules.

### DISCUSSION THREAD

**Discussion by gmencz** [vercel/next.js#20315] [Help]
# Bug report

## Describe the bug

I'm using a package (`sib-api-v3-typescript`) which does an import assignment in its source code:
```ts
import localVarRequest = require('request');
```
Because Next targets ES modules the compilation fails. I am able to run it in development if I set target to `commonjs` in my tsconfig but I cannot run the build (throws the error `Import assignment cannot be used when targeting ECMAScript modules. Consider using 'import * as ns from "mod"', 'import {a} from "mod"', 'import d from "mod"', or another module format instead.`).

## To Reproduce

1. Create a new typescript next project
2. `npm install sib-api-v3-typescript`
3. Use the installed library in your project

## Expected behavior

Successful compilation since it works during dev.

**Comment by polesapart** [comment]
I'm hit by the same package.

**Reply by martinratinaud** [reply]
Same for me, any news on that ?

**Comment by etodanik** [comment]
Same here, same package. Anyone has anything useful to share?

EDIT:
As per https://github.com/sendinblue/APIv3-typescript-library/issues/33
`2.2.3-beta.1` currently seems to fix this.

**Comment by phbernard** [comment]
Same here. [sib-api-v3-typescript](https://www.npmjs.com/package/sib-api-v3-typescript) has been deprecated, I really doubt this will be fixed.

**Reply by polesapart** [reply]
It was replaced by https://github.com/getbrevo/brevo-node - not sure if affected as I ended up using a custom-generated client.
